app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationServiceImpl.java

```diff
@@ -72,7 +78,26 @@ public ApplicationServiceImpl(Scheduler scheduler,
 
         return repository.findById(id, READ_APPLICATIONS)
                 .flatMap(this::setTransientFields)
-                .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.APPLICATION, id)));
+                .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.APPLICATION, id)))
+                .zipWith(ReactiveSecurityContextHolder.getContext())
```

Comment:
```suggestion
                .zipWith(SessionUserService.getCurrentUser())
```

This function will give you the `User` object directly. Easier to deal with across the codebase. Please make this change across the entire PR.

Reply:
@mohanarpit Thanks for the suggestion. I've updated accordingly.